### ISSUE TITLE
Google Two Factor Authentication breaks backup lockscreen method

### ISSUE THREAD

**Issue by pastudan** [mclear/Android_NFC_Ring_Unlock#38]
I have two factor authentication enabled on my Google account, so when I don't have my ring and try to use this as a backup authentication method, I have no way into my phone until I disable two factor auth from google (which might be impossible if i'm not signed into Google already, since the Authenticator app is on my phone)

**Comment by JohnMcLear** [comment]
Duplicate of https://github.com/mclear/Android_NFC_Ring_Unlock/issues/7 afaik...  @pastudan please confirm..

**Comment by pastudan** [comment]
Sorry, this is definitely a duplicate of #7 .. I just didn't look through the closed issues since it still seems like a persistent bug to me. Perhaps since we can't rely on Google authentication if 2-factor is enabled, make users enter a backup PIN or password before actually allowing them to enable this app.

**Comment by r2DoesInc** [comment]
I don't really know how 2 step author with Google works, so I don't really understand what the issue is. 

Can you explain exactly what the expected vs experienced behavior is?

**Comment by JohnMcLear** [comment]
I have seen others suffer with this issue, it probably needs some attention

**Comment by r2DoesInc** [comment]
I dont know if there is a way to resolve this issue, i think that requiring
the PIN will be a good way to get around this issue though, if the user has
a pin option and uses 2 step, the google auth wont work but theyll still
have something.

**Comment by theckman** [comment]
:+1: This problem bit me as well.

**Comment by shaferA** [comment]
Google expects to receive an app specific password based on the way the app is authenticating. These are generated by google and (I think) are only one time use. I was able to use them as a workaround, but this only works if I also have access to my google account already.

**Comment by pastudan** [comment]
Yup, it only works if you have already setup app-specific passwords. They
are not one time use, but you really shouldn't be writing these down
anywhere, as that defeats the purpose of Two Factor Authentication.

**Comment by MHauge** [comment]
+1
Would also like some other way that google-login to unlock.
2-factor auth gives a long list of random letters, which i can't remember - so need to store it somewhere, in case NFC fails to unlock.
This happened to me a few days ago, where another app on my phone caught the NFC, so phone didn't unlock, and when i got it unlocked, this app had started in the background.
